src/vineyard/views.py

```diff
@@ -23,45 +28,71 @@
 @csrf_exempt
 def index(request):
     """
-    Access database to respond with requested vineyard metadata.
+    Accesses database to respond with requested vineyard metadata.
     """
+
     if request.method != "POST":
-        return HttpResponseBadRequest()
+        return HttpResponseNotAllowed(['POST'])
 
     data = json.loads(request.body.decode('utf-8'))
-    vineyard_id = data.get('vineyard_id', '')
-    auth_token = data.get('auth_token', '')
-    response = {}
+    vineyard_id = str(data.get('vineyard_id', ''))
+    auth_token = str(data.get('auth_token', ''))
 
     try:
-        logger.info('Validating auth_token token for vineyard id {}.'.format(vineyard_id))
+        message = (
+            'Validating auth token for vineyard id {}.'
+        ).format(vineyard_id)
+        logger.info(message)
+
         cassy.verify_auth_token(auth_token)
+    except PlantalyticsException as e:
+        message = (
+            'Invalid auth token for vineyard id {}'
+        ).format(vineyard_id)
+        logger.warn(message)
+        error = custom_error(str(e))
+        return HttpResponseForbidden(error, content_type='application/json')
     except Exception as e:
-        message = 'Error occurred while auth token for vineyard id {}.'.format(vineyard_id)
-        logger.exception(message + '\n'+ str(e))
-        return HttpResponseForbidden()
+        message = (
+            'Error occurred while auth token for vineyard id {}{}{}.'
```

Comment:
I might do this as 'Error occurred while auth token for vineyard id {}\n{}.' unless for some reason that doesn't works.